modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -261,7 +261,7 @@ def split_partitions_using_pivots_for_sort(
         # When we only have one unique value in our pivots, `np.digitize` assumes that the pivots
         # are sorted in ascending order, and gives us results based off of that assumption - so if
         # we actually want to sort in descending order, we need to swap the new indices.
-        if not ascending and len(np.unique(pivots)) == 1 and len(pivots) != 1:
+        if not ascending and len(np.unique(pivots)) == 1:
             groupby_col = len(pivots) - groupby_col
```

Comment:
This branch is intended to inverse the order of the `digitize` result in order to comply with the descending order being set:
```python
>>> col_to_digitize = [1, 2, 3, 4]
>>> ascending = False
>>> pivots = [3.0, 3.0]
>>> groupby_col = np.digitize(col_to_digitize, pivots)
array([0, 0, 1, 1], dtype=int64)
>>> split_col_into_bins(col_to_digitize, groupby_col)
[[1, 2], [3, 4]] # these are in ascending order, however we wanted descending
>>> groupby_col = len(pivots) - groupby_col
array([1, 1, 0, 0], dtype=int64)
>>> split_col_into_bins(col_to_digitize, groupby_col)
[[3, 4], [1, 2]] # now they're in the order we wanted
```

At the first glance, there's nothing wrong with entering this branch if `len(pivots == 1)`. cc @RehanSD

Reply:
Tested on my end, and confirmed this is the case!